blueocean-pipeline-scm-api/src/main/java/io/jenkins/blueocean/credential/SSHCredentialManager.java

```diff
@@ -0,0 +1,188 @@
+        return key;
+    }
+    
+    /**
+     * Gets a readable SSH-compatible public key a user could paste somewhere
+     * @param user
+     * @param key 
+     * @return 
+     */
+    public String getReadablePublicKey(@Nonnull User user, BasicSSHUserPrivateKey key) {
+        PKCS8EncodedKeySpec keySpec =
+            new PKCS8EncodedKeySpec(Base64.decode(key.getPrivateKey()));
+        try {
+            KeyFactory keyFactory = KeyFactory.getInstance("RSA");
+            RSAPrivateCrtKey privateKey = (RSAPrivateCrtKey)keyFactory.generatePrivate(keySpec);
+            RSAPublicKeySpec publicKeySpec = new java.security.spec.RSAPublicKeySpec(privateKey.getModulus(), privateKey.getPublicExponent());
+            RSAPublicKey publicKey = (RSAPublicKey)keyFactory.generatePublic(publicKeySpec);
+            String id_rsa_pub = getSSHPublicKey(publicKey, user);
+            return id_rsa_pub;
+        } catch (InvalidKeySpecException | NoSuchAlgorithmException | IOException ex) {
+            throw new RuntimeException(ex);
+        }
+    }
+
+    /**
+     * Gets the user's CredentialStore
+     * @param user user to find a store for
+     * @return the credential store or null if not found
+     */
+    private static @CheckForNull CredentialsStore getUserStore(User user){
+        for(CredentialsStore s : CredentialsProvider.lookupStores(user)) {
+            if(s.hasPermission(CredentialsProvider.CREATE) && s.hasPermission(CredentialsProvider.UPDATE)){
+                return s;
+            }
+        }
+        return null;
+    }
+    
+    /**
+     * Gets an "ssh-rsa"-style formatted public key with useful identifier
+     */
+    private String getSSHPublicKey(RSAPublicKey key, User user) throws IOException {
+        return "ssh-rsa " + Base64.encode(encodePublicKey(key)) + " " + getKeyComment(user.getId());
+    }
+
+    /**
+     * Returns a <user>@<jenkins> sort of comment to help identify the key's origin
+     * @param userId a user id
+     * @return an identifier
+     */
+    private String getKeyComment(String userId) {
+        return ((userId == null ? "user" : userId) + "@"
+            + Jenkins.getInstance().getDisplayName())
+                .replaceAll("[^@._a-zA-Z0-9]", "");
+    }
+    
+    /**
+     * Generates a new RSA key with specified keySize
+     * @return a public/private key pair
+     */
+    private static KeyPair generateRSAKey(int keySize) {
+        try {
+            KeyPairGenerator generator = KeyPairGenerator.getInstance("RSA");
+            generator.initialize(keySize);
+            return generator.genKeyPair();
+        } catch (NoSuchAlgorithmException ex) {
+            throw new RuntimeException(ex);
+        }
+    }
+
+    /**
+     * Encodes the public key according to some spec somewhere
+     * @param key public key to use
+     * @return the ssh-rsa bytes
+     * @throws IOException 
+     */
+    private static byte[] encodePublicKey(RSAPublicKey key) throws IOException {
```

Comment:
perhaps `getReadablePublicKey` moves to a static on the key generation class? I think all the crypto usage shouldn't be in this class.